src/libs/TransactionUtils.ts

```diff
@@ -0,0 +1,358 @@
+import Onyx, {OnyxCollection} from 'react-native-onyx';
+import {format, parseISO, isValid} from 'date-fns';
+import CONST from '../CONST';
+import ONYXKEYS from '../ONYXKEYS';
+import DateUtils from './DateUtils';
+import * as NumberUtils from './NumberUtils';
+import {RecentWaypoints, ReportAction, Transaction} from '../types/onyx';
+import {Receipt, Comment} from '../types/onyx/Transaction';
+import {OriginalMessageIOU} from '../types/onyx/OriginalMessage';
+
+type TransactionChanges = Partial<Transaction> & {comment?: string};
```

Comment:
Think about extracting this merged object to a separate type - it might grow over time.

Reply:
Yep, we are working on extending money requests and split bills. So, we are adding categories, tags and so on.

Reply:
@fabioh8010 `transactionChanges` stores only the comment text and not the whole comment. This text is then put inside the updated transaction comment. 

See this code snippet

    if (Object.hasOwn(transactionChanges, 'comment')) {
        updatedTransaction.comment = {
            ...updatedTransaction.comment,
            comment: transactionChanges.comment,
        };
    }